pkg/controller/machine.go

```diff
@@ -248,7 +248,7 @@ func (c *controller) getMachineFromNode(nodeName string) (*v1alpha1.Machine, err
 }
 
 func (c *controller) updateMachineState(machine *v1alpha1.Machine, node *v1.Node) error {
-	machine = c.updateMachineConditions(machine, node.Status.Conditions)
+	machine, _ = c.updateMachineConditions(machine, node.Status.Conditions)
```

Comment:
Should we print a warning if updateMachineCondition is returning an error?

Reply:
Ideally, we should do proper error handling. Should we even proceed in case of errors, needs similar changes in other methods as well, thought of doing this together for other methods as well. Ok with you?

Reply:
Sounds good.